pkg/authorization/registry/subjectaccessreview/rest.go

```diff
@@ -50,21 +50,39 @@ func (r *REST) Create(ctx kapi.Context, obj runtime.Object) (runtime.Object, err
 		return nil, err
 	}
 
-	var userToCheck user.Info
+	var userToCheck *user.DefaultInfo
 	if (len(subjectAccessReview.User) == 0) && (len(subjectAccessReview.Groups) == 0) {
 		// if no user or group was specified, use the info from the context
 		ctxUser, exists := kapi.UserFrom(ctx)
 		if !exists {
 			return nil, kapierrors.NewBadRequest("user missing from context")
 		}
-		userToCheck = ctxUser
+		// make a copy, we don't want to risk changing the original
+		userToCheck = &user.DefaultInfo{
+			Name:   ctxUser.GetName(),
+			Groups: ctxUser.GetGroups(),
+			UID:    ctxUser.GetUID(),
+			Extra:  ctxUser.GetExtra(),
```

Comment:
Need to copy the map or the `Extra` map held by ctxUser is vulnerable to mutation below